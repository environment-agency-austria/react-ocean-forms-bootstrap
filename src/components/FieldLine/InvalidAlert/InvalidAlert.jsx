/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faExclamationCircle from '@fortawesome/fontawesome-free-solid/faExclamationCircle';

/**
 * Displays an error icon if valid is false
 */
function InvalidAlert({ valid }) {
  return !valid && <FontAwesomeIcon icon={faExclamationCircle} className="ico-invalid mr-2" />;
}

InvalidAlert.propTypes = {
  valid: PropTypes.bool,
};

export default InvalidAlert;
