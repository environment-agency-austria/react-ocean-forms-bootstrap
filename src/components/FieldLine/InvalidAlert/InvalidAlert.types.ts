/**
 * Props for the InvalidAlert component
 */
export interface IInvalidAlertProps {
  /**
   * True if the field is valid and thus the
   * component should not be rendered
   */
  valid: boolean;
}
