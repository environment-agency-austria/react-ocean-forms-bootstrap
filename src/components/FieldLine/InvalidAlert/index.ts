export * from './InvalidAlert';
export * from './InvalidAlert.types';
