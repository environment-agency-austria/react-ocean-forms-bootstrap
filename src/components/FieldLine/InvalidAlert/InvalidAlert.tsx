/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import * as React from 'react';

import { faExclamationCircle } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { IInvalidAlertProps } from './InvalidAlert.types';

/**
 * Displays an error icon if valid is false
 */
export class InvalidAlert extends React.Component<IInvalidAlertProps> {
  public static displayName: string = 'InvalidAlert';

  public render(): JSX.Element | null {
    const { valid } = this.props;

    if (valid) { return null; }

    return (
      <FontAwesomeIcon
        icon={faExclamationCircle}
        className="ico-invalid mr-2"
      />
    );
  }
}
