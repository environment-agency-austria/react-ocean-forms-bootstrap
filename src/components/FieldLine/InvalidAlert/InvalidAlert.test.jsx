import React from 'react';
import { shallow } from 'enzyme';

import { InvalidAlert } from './InvalidAlert';

describe('<InvalidAlert />', () => {
  const setup = valid => shallow((
    <InvalidAlert
      valid={valid}
    />
  ));

  it('should render nothing if valid is true', () => {
    const wrapper = setup(true);
    expect(wrapper.exists('FontAwesomeIcon')).toBeFalsy();
  });

  it('should render a FontAwesomeIcon if valid is false', () => {
    const wrapper = setup(false);
    expect(wrapper.exists('FontAwesomeIcon')).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
