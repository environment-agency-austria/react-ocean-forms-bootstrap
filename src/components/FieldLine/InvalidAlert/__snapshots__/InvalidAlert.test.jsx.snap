// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InvalidAlert /> should render a FontAwesomeIcon if valid is false 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InvalidAlert
    valid={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "border": false,
      "className": "ico-invalid mr-2",
      "fixedWidth": false,
      "flip": null,
      "icon": Object {
        "icon": Array [
          512,
          512,
          Array [],
          "f06a",
          "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
        ],
        "iconName": "exclamation-circle",
        "prefix": "fas",
      },
      "inverse": false,
      "listItem": false,
      "mask": null,
      "pull": null,
      "pulse": false,
      "rotation": null,
      "size": null,
      "spin": false,
      "symbol": false,
      "title": "",
      "transform": null,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "border": false,
        "className": "ico-invalid mr-2",
        "fixedWidth": false,
        "flip": null,
        "icon": Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f06a",
            "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
          ],
          "iconName": "exclamation-circle",
          "prefix": "fas",
        },
        "inverse": false,
        "listItem": false,
        "mask": null,
        "pull": null,
        "pulse": false,
        "rotation": null,
        "size": null,
        "spin": false,
        "symbol": false,
        "title": "",
        "transform": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
